### ISSUE TITLE
Remove dependency on unique nick names

### ISSUE THREAD

**Issue by blenderskool** [blenderskool/blaze#72]
The unique nicknames rule was originally added to prevent multiple sessions from the same browser window. It is used in some places to identify a peer in the room. This dependency on the unique nicknames should be removed and **auto-generated unique id must be used**.

**Comment by omnone** [comment]
Hey I am interested on this, can you provide some more information ? Thanks!

**Comment by blenderskool** [comment]
Hi @omnone,
A quick way to see the issue is by going to Blaze from two different tabs and join the same room. There will be a message stating `User with the same name exists in this room`. This is because Blaze expects the name of the user to be unique in a room.

The best place to start working on this issue is to:
- go to `socketConnect.js` and check what data we are sending right now while joining a room. Maybe add an auto-generated id here.
- check the WebSocket **server** in `server/index.js` and update it to use the above id instead of `name` in some places. Especially in `room.broadcast` method.
- Update the frontend to make use of the above id too. _Most of the code_ in the frontend uses `name` or `peerId` for identifying the peer right now, All of these need to be refactored accordingly.

All in all, this issue has a large impact area.

**Comment by omnone** [comment]
Awesome i will give it a try thanks!

**Comment by blenderskool** [comment]
Sure @omnone :smile:

**Comment by omnone** [comment]
Will users be still able to select their usernames or will uuids be used everywhere?

**Comment by blenderskool** [comment]
@omnone yes users will still be able to use nicknames. The flow does not change for a user in anyway. Only internally users would be identified by the unique id
